merchantapi/src/main/java/fi/hel/verkkokauppa/configuration/service/MerchantService.java

```diff
@@ -57,6 +73,46 @@ public MerchantDto save(MerchantDto dto) {
         return merchantMapper.toDto(saved);
     }
 
+    public String encryptSecret(String secret) {
+        String salt = env.getRequiredProperty("merchant.secret.encryption.salt");
+        return encryptService.encryptSecret(secret, salt);
+    }
+
+    public String decryptSecret(String secret) {
+        String salt = env.getRequiredProperty("merchant.secret.encryption.salt");
+        return decryptService.decryptSecret(salt, secret);
+    }
+
+    public PaytrailMerchantMappingDto save(PaytrailMerchantMappingDto dto) {
+        PaytrailMerchantMapping entity = paytrailMerchantMappingMapper.fromDto(dto);
+        entity.setMerchantPaytrailSecret(encryptSecret(entity.getMerchantPaytrailSecret()));
+        entity.setId(UUIDGenerator.generateType3UUIDString(entity.getNamespace(), entity.getMerchantPaytrailMerchantId()));
+        entity = paytrailMerchantMappingRepository.save(entity);
+        entity.setMerchantPaytrailSecret(decryptSecret(entity.getMerchantPaytrailSecret()));
+        return paytrailMerchantMappingMapper.toDto(entity);
+    }
+
+    public PaytrailMerchantMappingDto getPaytrailMerchantMappingByMerchantPaytrailMerchantIdAndNamespace(
+        String merchantPaytrailMerchantId,
+        String namespace
+    ) {
+        String id = UUIDGenerator.generateType3UUIDString(namespace, merchantPaytrailMerchantId);
+        if (id == null) {
+            throw new CommonApiException(
+                    HttpStatus.INTERNAL_SERVER_ERROR,
+                    new Error("failed-to-get-paytrail-merchant-mapping", "failed to generate id for mapping with namespace, merchantPaytrailMerchantId: [" + namespace + ", " + merchantPaytrailMerchantId + "]")
+            );
+        }
+        PaytrailMerchantMapping mapping = paytrailMerchantMappingRepository
+                .findById(id)
+                .orElseThrow(() -> new CommonApiException(
+                        HttpStatus.NOT_FOUND,
+                        new Error("paytrail-merchant-mapping-not-found", "paytrail merchant mapping with namespace, merchantPaytrailMerchantId: [" + namespace + ", " + merchantPaytrailMerchantId + "] not found")
+                ));
+        mapping.setMerchantPaytrailSecret(decryptSecret(mapping.getMerchantPaytrailSecret()));
```

Comment:
Create function for this to make this clear why this is used. -> like decryptPaytrailSecret etc